Spanner/src/Session/CacheSessionPool.php

```diff
@@ -313,6 +314,9 @@ public function acquire($context = SessionPoolInterface::CONTEXT_READ)
 
         // If we don't have a session, let's wait for one or throw an exception.
         if (!$session) {
+            if (!is_null($exception)) {
+                throw $exception;
```

Comment:
There are 2 things that you might want to validate:
1. Is there any scenario when an exception is returned by `createSessions(..)` even when atleast one session is created?
2. In real scenario, would the user get exception with message `error`? If so, better wrap it into LogicException like below in `maintain(..)` function:

```
throw new \LogicException('Cannot maintain session pool: database not set.');
```

Reply:
1. The `createSessions` uses `batchCreateSessions` to create the required number of sessions. If you see here https://github.com/googleapis/google-cloud-php/blob/4e77c34b73b76776f6dd1bddb1b02a31e393d417/Spanner/src/Session/CacheSessionPool.php#L691, they break in the event of an exception and come out with empty session array. So, exception will be thrown with empty session list by the `createSessions` function.
2. As per @bshaffer comment, we changed it to `RuntimeException`

Reply:
makes sense to me now.. thanks for clarification.